Test multiple jwt versions

zubin opened the pull request:
## Context

Currently the test suite fails when tested against JWT gem < v2. 

## Change

- Update build to test against multiple JWT gem versions (latest minor versions since 1.5.x). 
- Resolve regression against JWT < v2. 

For further details, see the new _Compatibility_ section of the README. 

Since there's no change in behaviour, there's no need to create a new version.

zubin committed:
Sort

zubin committed:
Formatting

zubin committed:
Test against multiple jwt versions

zubin committed:
Document how to test multiple JWT versions

zubin committed:
Formatting

zubin committed:
Target leeway in assertion

orien commented on gemfiles/jwt_1.5.6.gemfile.lock:
```diff
@@ -0,0 +1,55 @@
+PATH
```
Perhaps we don't need to check in the lock files. Do we mind if gems other than `jwt` get upgraded?

zubin replied:
I was thinking that too but appraisal's readme says:

> ... These are stored in the gemfiles directory, and should be added to version control to ensure that the same versions are always used.

https://github.com/thoughtbot/appraisal#under-the-hood

Unlike `Gemfile.lock`, it doesn't affect applications which depend on it.

orien replied:
I also see [this](https://github.com/thoughtbot/appraisal#version-control).

> When using Appraisal, we recommend you check in the Gemfiles that Appraisal generates within the gemfiles directory, but exclude the lockfiles there (`*.gemfile.lock`.)

zubin replied:
Ah ok cool. Updated: 93fe319

orien reviewed (approved):
👏 🎉  Awesome! Thank you.

zubin committed:
Resolve CI failure

    You must use Bundler 2 or greater with this lockfile.
    The command "bundle exec appraisal rspec" exited with 1.

zubin committed:
Ensure required bundler version is installed

zubin committed:
Sort

zubin committed:
Exclude appraisal gemfile.lock files
